src/config/experiment.js

```diff
@@ -1,22 +1,64 @@
-import { defaultBlockSettings } from './main';
-import { deepCopy } from '../lib/utils'
+import { deepCopy } from "../lib/utils"
+import localConfig from "./config.json"
+import { envConfig } from "./main"
+import path from "path"
+import { firestoreConfig } from "../firebase"
 
-// FIRST EXPERIMENT BLOCK SETTINGS
+const getConfig = async (participantID, studyID) => {
+  let experimentConfig = localConfig
+  if (envConfig.USE_ELECTRON) {
+    const app = window.require("electron").remote.app
+    const renderer = window.require("electron").ipcRenderer
+    const fs = window.require("fs")
+    try {
+      const overrideConfig = path.join(
+        app.getPath("desktop"),
+        `${app.getName()}-settings`,
+        `${participantID}-config.json`
+      )
+      experimentConfig = Object.assign(experimentConfig, JSON.parse(fs.readFileSync(overrideConfig), "utf8"))
+    } catch (error) {
+      console.warn("Using default config")
+    }
+    renderer.send("save-config", experimentConfig, participantID, studyID)
+  } else if (envConfig.USE_FIREBASE) {
+    const newConfig = await firestoreConfig(studyID, participantID);
```

Comment:
maybe `participantConfig` or `firestoreConfig` (once function is renamed) instead of `newConfig`